resource/file/fetch/fetch.go

```diff
@@ -0,0 +1,217 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetch
+
+import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"io"
+	"net/url"
+	"os"
+
+	"github.com/asteris-llc/converge/resource"
+	"github.com/hashicorp/go-getter"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// Hash type for Fetch
+type Hash string
+
+const (
+	// HashMD5 indicates hash type md5
+	HashMD5 Hash = "md5"
+
+	// HashSHA1 indicates hash type sha1
+	HashSHA1 Hash = "sha1"
+
+	// HashSHA256 indicates hash type sha256
+	HashSHA256 Hash = "sha256"
+
+	// HashSHA512 indicates hash type sha512
+	HashSHA512 Hash = "sha512"
+)
+
+// Fetch gets a file and makes it available on disk
+type Fetch struct {
+	Source      string
+	Destination string
+	HashType    string
+	Hash        string
+	Force       bool
+
+	hasApplied bool
+}
+
+// Check if the the file is on disk, and the hashes are availabe
+func (f *Fetch) Check(context.Context, resource.Renderer) (resource.TaskStatus, error) {
+	var (
+		hsh    hash.Hash
+		err    error
+		status = resource.NewStatus()
+	)
+
+	if !f.hasApplied {
```

Comment:
Could you do `if f.hasApplied { return status, nil }` here instead to reduce nesting of the rest of the code?